### PR TITLE
Support locally including private sample animations in example app and regression suite

### PR DESCRIPTION
**Pull request by calda**
This PR adds support for locally including private sample animations in the example app and regression suite.

Any sample animations stored locally in `/Tests/Samples/Private` will be displayed in the example app and tested as a part of the regression suite, but are `.gitignore`'d so will not be included in the public repo.

This screenshot shows the Example app displaying some animations that we use internally at Airbnb:

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -20,5 +20,5 @@ let package = Package(
       dependencies: ["Lottie", "SnapshotTesting"],
       path: "Tests",
       exclude: ["Artifacts"],
-      resources: [.process("Samples"), .process("__Snapshots__")]),
+      resources: [.copy("Samples"), .process("__Snapshots__")]),
```
Comment (calda) [inline Package.swift:23]: I updated the snapshot tests to include the full path of the sample json file in the snapshot name.

This was necessary to `.gitignore` snapshot of samples in the `Samples/Private` folder, and also removes the potential for naming conflicts that could happen previously.
 
 To do this, we have to change the resource strategy from `process` (which takes all of the resource files and just includes them directly in the bundle's root folder and doesn't preserve any subdirectory information) to `copy` (which copies the exact directory structure).

File: Tests/SnapshotTests.swift [hunk 2]
```diff
@@ -63,7 +75,23 @@ class SnapshotTests: XCTestCase {
...
   /// The list of sample animation files in `Tests/Samples`
-  private let sampleAnimationURLs = Bundle.module.urls(forResourcesWithExtension: "json", subdirectory: nil)!
+  private lazy var sampleAnimationURLs: [URL] = {
+    let enumerator = FileManager.default.enumerator(atPath: Bundle.module.bundlePath)!
+
+    var sampleAnimationURLs: [URL] = []
+
+    while let fileSubpath = enumerator.nextObject() as? String {
+      if
+        fileSubpath.hasPrefix("Samples"),
```
Comment (erichoracek) [inline Tests/SnapshotTests.swift:85]: Extract out into a shared constant with line 136?

**Review by erichoracek — APPROVED** [review]
Good stuff! Nice functionality to add

File: Tests/SnapshotTests.swift [hunk 3]
```diff
@@ -34,11 +34,21 @@ class SnapshotTests: XCTestCase {
   /// a sample JSON file that is visible to this test target.
   func testAllSnapshotsHaveCorrespondingSampleFile() {
     for snapshotURL in snapshotURLs {
-      // The snapshot files follow the format `testLottieSnapshots.NAME-PERCENTAGE.png`
-      let animationName = snapshotURL.lastPathComponent.components(separatedBy: .init(charactersIn: ".-"))[1]
+      // The snapshot files follow the format `testCaseName.animationName-percentage.png`
+      //  - We remove the known prefix and known suffixes to recover the input file name
+      //  - `animationName` can contain dashes, so we can't just split the string at each dash
+      let animationName = snapshotURL.lastPathComponent
+        .replacingOccurrences(of: "testExperimentalRenderingEngine.", with: "")
```
[the remark this answers is not in the document]
Comment (calda) [reply Tests/SnapshotTests.swift:41]: Good idea!

[the remark this answers is not in the document]
Comment (calda) [reply Tests/SnapshotTests.swift:41]: Looks like the objc selector for these methods follows a different format:

```
(lldb) p #selector(testExperimentalRenderingEngine)
(ObjectiveC.Selector) $R2 = "testExperimentalRenderingEngineAndReturnError:"

(lldb) p #selector(testLottieSnapshots)
(ObjectiveC.Selector) $R3 = "testLottieSnapshotsAndReturnError:"
```

**Pull request merged**